app/src/main/java/com/github/sdpcoachme/schedule/ScheduleActivity.kt

```diff
@@ -96,103 +127,173 @@ private class EventDataModifier(val event: ShownEvent) : ParentDataModifier {
     override fun Density.modifyParentData(parentData: Any?) = event
 }
 
+private const val ColumnsPerWeek = 7
 @Composable
 fun Schedule(
     futureUserInfo: CompletableFuture<UserInfo>,
     modifier: Modifier = Modifier,
 ) {
-    // bind those to database
     var events by remember { mutableStateOf(emptyList<Event>()) }
     var eventsFuture by remember { mutableStateOf(futureUserInfo.thenApply { it.events }) }
+    val context = LocalContext.current
 
     LaunchedEffect(eventsFuture) {
         eventsFuture.thenAccept { e ->
             if (e != null) {
                 events = e
-                val f = CompletableFuture<List<Event>?>()
-                f.complete(null)
-                eventsFuture = f
+                eventsFuture = CompletableFuture.completedFuture(null)
             }
         }.exceptionally {
-            //TODO in next sprint: handle error
-            /*val errorMsg = "Schedule couldn't get the user information from the database." +
+            val errorMsg = "Schedule couldn't get the user information from the database." +
                 "\n Please return to the login page and try again."
-            ErrorHandlerLauncher().launchExtrasErrorHandler(context, errorMsg)*/
+            ErrorHandlerLauncher().launchExtrasErrorHandler(context, errorMsg)
             null
         }
     }
 
-    // the starting day is always the previous Monday
-    val minDate = LocalDate.now().with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
-    val maxDate = minDate.plusDays(ColumnsPerWeek.toLong())
     val dayWidth = LocalConfiguration.current.screenWidthDp.dp / ColumnsPerWeek
-    val hourHeight = 64.dp
     val verticalScrollState = rememberScrollState()
-    val horizontalScrollState = rememberScrollState()
-    Column(modifier = modifier.testTag(ScheduleActivity.TestTags.SCHEDULE_COLUMN)) {
-        ScheduleHeader(
-            minDate = minDate,
-            maxDate = maxDate,
+    // the starting day is always the monday of the current week
+    val startMonday = LocalDate.now().with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY))
+
+    var shownWeekMonday by remember { mutableStateOf(startMonday) }
+
+    fun updateCurrentWeekMonday(weeksToAdd: Int) {
```

Comment:
Great idea! This keeps us flexible to add features for skipping multiple weeks at a time (e.g., if we decide to add a go-to-current-week button, etc.)